cpp/src/arrow/compute/kernels/scalar_validity.cc

```diff
@@ -337,16 +333,15 @@ const FunctionDoc is_nan_doc("Return true if NaN",
 void RegisterScalarValidity(FunctionRegistry* registry) {
   static auto kNullOptions = NullOptions::Defaults();
   MakeFunction("is_valid", is_valid_doc, {ValueDescr::ANY}, boolean(), IsValidExec,
-               registry, MemAllocation::NO_PREALLOCATE, NullHandling::OUTPUT_NOT_NULL,
```

Comment:
Note: is_valid will no longer do zero copy optimizations. I think these buffer-reuse (whether slicing the input buffer or passing it on to the output) optimizations have relatively little value on the whole because they prevent parallel execution by input-splitting (the ideal scenario for parallel execution is to have each parallel task writing into a slice of a larger memory preallocation)